src/System.Security.Cryptography.Pkcs/src/Internal/Cryptography/Pal/Unix/PkcsPalOpenSsl.Encrypt.cs

```diff
@@ -37,26 +44,34 @@ internal sealed partial class PkcsPalOpenSsl : PkcsPal
 
                     ClassifyAndAddRecipients(cms, recipients);
 
-                    foreach (X509Certificate2 cert in originatorCerts)
+                    if (contentInfo.ContentType.Value != Oids.Pkcs7Data)
                     {
-                        using (SafeX509Handle certHandle = Interop.Crypto.X509Duplicate(cert.Handle))
+                        using (SafeAsn1ObjectHandle oid = Interop.Crypto.ObjTxt2Obj(contentInfo.ContentType.Value))
                         {
-                            Interop.Crypto.CheckValidOpenSslHandle(certHandle);
-                            status = Interop.Crypto.CmsAddOriginatorCert(cms, certHandle);
```

Comment:
If we're going to say that originator certs just aren't supported we should delete the shim function and the extern method.